.github/workflows/basic.yml

```diff
@@ -50,4 +50,4 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: --all-targets --all-features -- -D warnings
\ No newline at end of file
+          args: --all-targets --all-features
```

Comment:
TBH I can't even reproduce the `must_use` errors, even with a nightly toolchain (I can reproduce the `.to_string()` error though). But with the added `must_use`s, is it really necessary to allow warnings?

Reply:
After rustup-nightly was updated to version 1.59.0,
the lint check warns when a method returning Self doesn’t have the #[must_use] attribute.
(refer to: https://rust-lang.github.io/rust-clippy/master/index.html#return_self_not_must_use)
However, at present, the rust source code generated by proto built by tonic tool chain
does not comply with this requirement, which will lead to CI/CD running failure: 
https://github.com/confidential-containers/attestation-agent/runs/4961352636?check_suite_focus=true#step:8:110

Reply:
Hm. Is it important we run on nightly? Because I should think tonic will have fixed this by the time 1.59 becomes stable, and then we won't have to think about re-enabling `-D warnings` (or fix additional warnings that come up by then).